webdriver-spec.html

```diff
@@ -3237,183 +3237,428 @@ <h2 id=element-displayedness>Element Displayedness</h2>
+
+      <li><p>Let <var>script</var>'s <a href="">code entry-point</a>
+            be <var>function</var></p></li>
+
+      <li><p>Let script's <a href="">settings object</a> object
+          be <var>script settings</var>.</p></li>
+
+      <!-- the stuff from here on down is quite dubious -->
+      <!-- this incumbent scripts stuff is in webidl but the references don't
+           lead anywhere -->
+      <!-- li><p><a href="">Push <var>script</var> onto <a>the stack of
+              incumbent scripts</a>.</li-->
+
+      <li><p>Invoke the <code>[[\Call]]</code> internal method
+          of <var>function</var>, providing <var>window</var> as the
+          this value and <var>parameters</var> as the argument
+          values. If doing so does not produce an exception,
+          let <var>result</var> result be <a>success</a> with data set
+          to the return value from this function call. Otherwise
+          let <var>result</var> be <a>error</a> with
+          code <a>javascript error</a>.</p>
+
+      <!-- li><p><a href="">Pop <var>script</var> from <a>the stack of
+              incumbent scripts</a>.</p></li -->
+
+      <li><p>If <var>result</var> is an <a>error</a>, return
+          result.</p></li>
+
+      <li><p>Otherwise let <var>json data</var> be a <a>JSON clone</a>
+          of <var>result</var>'s data.</p></li>
+
+      <li><p>Return <a>success</a> with data <var>json
+            data</var>.</p></li>
+
+    </ol>
+
+  <section>
+    <h3>Execute Script</h3>
+    <p>
+      <table class="simple jsoncommand">
+        <tr>
+          <th>HTTP Method</th>
+          <th>Path Template</th>
+          <th>Notes</th>
+        </tr>
+        <tr>
+          <td>POST</td>
+          <td id='post-execute_script'>/session/{sessionId}/execute</td>
+          <td></td>
+        </tr>
+      </table>
+    <p>The <dfn>Execute Script</dfn> command executes a javascript
+      function in the context of the <a>current browsing context</a>
+      and return the return value of the function.
+
+    <p>The <a>remote end steps</a> for the <a>Execute Script</a>
+        command are:</p>
+    <ol>
+      <li><p>Let <var>script arguments</var> be the result
+      of <a>extracting the script arguments from a request</a> with
+      argument <var>parameters</var>.</p></li>
+
+      <li><p>If <var>script arguments</var> is an <a>error</a>
+      return <var>script arguments</var>.</p></li>
+
+      <li><p>Let <var>body</var> and <var>arguments</var>
+      be <var>script arguments</var>' data.</p></li>
+
+      <li><p>Let <var>result</var> be the result of calling <a>execute
+      a function body</a>, with arguments <var>body</var>
+      and <var>arguments</var>.</p></li>
+
+      <li><p>If <var>result</var> is an <a>error</a>, return result,
+      otherwise let <var>value</var> be <var>result</var>'s data</p></li>
+
+      <li><p>Let <var>data</var> be a new <code>Object</code></p></li>
```

Comment:
I notice you’ve started using `<code>Object</code>` instead of JSON Object.  I think this makes sense, but just to clarify, are we then using internal JS data structures which our serialisation definitions will convert into JSON Object’s on their way out?